crypto/evp/evp_enc.c

```diff
@@ -1055,6 +1062,7 @@ int EVP_CIPHER_CTX_ctrl(EVP_CIPHER_CTX *ctx, int type, int arg, void *ptr)
     switch (type) {
     case EVP_CTRL_SET_KEY_LENGTH:
         params[0] = OSSL_PARAM_construct_size_t(OSSL_CIPHER_PARAM_KEYLEN, &sz);
+        ctx->key_len = -1;
```

Comment:
Is this ok to set this to -1 for the legacy path?

Reply:
Yes, just above is a check for the legacy path and this code isn't executed in that case.